uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Helper/NavigationMenuHelper.php

```diff
@@ -94,23 +78,47 @@ public static function relocateAfter(
     $itemToMovePath,
     $precedingItemPath
   ) {
-    $siblings = &self::getSiblingsReference($menu, $precedingItemPath);
+    self::relocateWithOffset($menu, $itemToMovePath, $precedingItemPath);
+  }
+
+  /**
+   * @param $menu
+   *   The full menu structure
+   * @param $itemToMovePath
+   *   The path to the menu item we want to move
+   * @param $targetSiblingPath
+   *   The path to the sibling the item will be moved next to
+   * @param int $offset
+   *   How many places the item should be before / after the sibling item. Use
+   *   0 to place it right after it.
+   */
+  private static function relocateWithOffset(
+    &$menu,
+    $itemToMovePath,
+    $targetSiblingPath,
+    $offset = 0
```

Comment:
It would seem more consistent if 1 as the offset value will move the element after the $targetSiblingPath like -1 moves it just before it.

Reply:
It does seem like that, but then I'd wonder what the expected behaviour would be for an offset of 0. Although I agree that 1 for "right after" and -1 for "right before" might be clearer, it would have this empty hole in the middle of the scale.

|-1|0|1|
|--|--|--|
|1 before target| after target| two after target|

OR

|-1|0|1|
|--|--|--|
|1 before target| ? | after target|

After thinking a bit about this I think maybe you're right. The definition for offset in programming is:

> an offset within an array or other data structure object is an integer indicating the distance between the beginning of the object and a given element or point

If the distance between the target and item to insert is 0 it would seem that it would replace it. Obviously we don't want a method called `relocate..` to replace anything, so maybe an exception if offset of 0 is provided would be best.

Reply:
Yeah, would be a good idea to throw an exception when an offset of 0 is supplied.